### ISSUE TITLE
Sonobus Linux - Memory/Loop leak?

### ISSUE THREAD

**Issue by jcrystal** [essej/sonobus#9]
Debian stretch / JACK (dummy device)

Sometimes, after running successfully for several minutes, the Linux client will disconnect (leaving other devices to see it as "muted").

In this state, the SonoBus process spins up to 100%+ of CPU usage stops processing audio, and drags down the CPU to a grind. Only way out seems to be to manually SSH/kill that process. Possibly caused by a process XRun in Jack, but tough to determine. I can't identify what causes this to happen, but it seems to be anytime from 10 seconds to 5 minutes after starting a (successful) session.

Any way to help by extracting a log file from somewhere?

Thank you!

**Comment by essej** [comment]
There wouldn't be anything logged by default... have you tried it with a "real" audio interface and had this happen? When you look at 'top' output when it is in this state, is it just the CPU usage that has spiked, or is there a change in memory usage?

**Comment by jcrystal** [comment]
Yeah, I think a "real" audio interface is the next step. I'm attempting to get this rolling on an AWS instance, where I obviously don't have access to physical hardware. I wonder if there is a more stable virtual audio interface out there...

But to answer your question, while the CPU usage spikes, but the %MEM stays fairly constant. Really seems to just be CPU.

**Comment by essej** [comment]
Just a question, but why would you want to use this on an AWS instance? Currently it is using a pure peer-to-peer network setup, the only place you need to run it is at the endpoints where humans and audio interfaces are. In the future, there might be a server-based setup that would allow for a non-peer-to-peer-only option for mixing and/or to act as a relay... but the current version doesn't have that capability right now.

**Comment by jcrystal** [comment]
That situation you describe is exactly what I was building out; an AWS instance (not a peer) acting as a peer and routing to Jack (and returning). My guess is the lack of "real" hardware is what's causing it to take a dive, sadly.

I'll let you know if I come up with a solution on that front... :)

**Comment by JackCahn** [comment]
I am having the same symptom, but my CPU never recovers (everything is locked up until hard reset.)  Running Ubuntu studio 20.10.  I am connected only 3-5 seconds.

**Comment by JackCahn** [comment]
Additional info:  Using USB Focusrite Scarlet.  Jack settings with qjackctl USB, 48000, 128, 2.   Audio settings both input and output set to use jack connection kit.  Mic meter and test tone both work.  I hear participants clearly for a few seconds.
